### DISCUSSION TITLE
Add Attachemnts API for partial read

### DISCUSSION THREAD

**Discussion by harrison314** [ravendb/ravendb#14661] [Ideas]
I think RevanDb lacks an API that allows you to read/download the attachment in chunks.

eg: `session.Advanced.Attachments.GetChunk("albums/1", "document01.pdf", offset:0, length:150)`

Such an API solves an issue where it is necessary to implement seekable `Stream` for attachments.

For example, this should be used in DMS, where the file type is detected using the first 3 bytes. Or when parsing text from large PDFs (the entire file must be downloaded to the application).

**Comment by garayx** [comment]
Hi 

Thanks for suggestion, I have added this to our ticket tracker:
https://issues.hibernatingrhinos.com/issue/RavenDB-19049/Add-support-for-seekable-attachments

Currently only downloading entire file is supported.
